enhancements/sig-architecture/64-placementStrategy/README.md

```diff
@@ -470,15 +470,28 @@ type RolloutStrategy struct {
 	ProgressivePerGroup *RolloutProgressivePerGroup `json:"progressivePerGroup,omitempty"`
 }
 
-// Timeout to consider while applying the workload.
-type Timeout struct {
-	// Timeout define how long workload applier controller will wait till workload reach successful state in the cluster. Only considered for Rollout Type Progressive and ProgressivePerGroup.
-	// Timeout default value is None meaning the workload applier will not proceed apply workload to other clusters if did not reach the successful state.
-	// Timeout must be defined in [0-9h]|[0-9m]|[0-9s] format examples; 2h , 90m , 360s
+// RolloutConfig contains common configurations for the rollout strategies.
+type RolloutConfig struct {
+	// MinSuccessTime is a "soak" time. In other words, the minimum amount of time the workload applier controller will 
+  // wait from the start of each rollout before proceeding (assuming a successful state has been reached and MaxFailures wasn't breached).
+	// The default value is 0 meaning the workload applier proceed immediately after a successful state is reached.
+	// MinSuccessTime must be defined in [0-9h]|[0-9m]|[0-9s] format examples; 2h , 90m , 360s
+	// +kubebuilder:default:=0
+	// +optional
+	MinSuccessTime metav1.Time `json:"minSuccessTime,omitempty"`
+  // ProgressDeadline defines how long the workload applier controller will wait until workload reaches a successful 
+  // state in the cluster, marking it as failed after the deadline.
+	// The default value is None meaning the workload applier will continue to wait for a status if did not reach the successful state.
+	// ProgressDeadline must be defined in [0-9h]|[0-9m]|[0-9s] format examples; 2h , 90m , 360s
 	// +kubebuilder:validation:Pattern="^(([0-9])+[h|m|s])|None$"
 	// +kubebuilder:default:=None
 	// +optional
-	Timeout string `json:"timeout,omitempty"`
+  ProgressDeadline string `json:"progressDeadline,omitempty"`
+  // MaxFailures is a percentage of the current rollout that can fail before proceeding to the next rollout.
+	// Default is that no failures are tolerated.
+  // +kubebuilder:validation:Maximum=100
+	// +optional
+  MaxFailedPercent uint8 `json:"maxFailedPercent,omitempty"`
```

Comment:
I believe the name should be **maxFailures**  and better to have type as intstr.IntOrString to allow percentage and numbers with validation as below.
`// +kubebuilder:validation:Pattern="^((100|[0-9]{1,2})%|[0-9]+)$"`

Reply:
Would that be similar to `ClustersPerDecisionGroup` where it's either a number of clusters or a percentage of clusters? If so, that sounds good to me.

Reply:
yes, similar to ClustersPerDecisionGroup